### PR TITLE
Musical instruments

### PR DESCRIPTION
**Pull request by FluffyGhoster**
The final run for the implementation of musical instruments.

This PR implements the musical instruments using (mostly) a sound synthesizer, as ported from Baystation 12, capitalizing on the sound tokens and player from my previous PRs.

The guitar, violin and trumpeth can spawn in maintenance loots and be found from traders.

The piano and synthesizer (the actual instrument/object) will be orderable from cargo after the DB is updated accordingly to allow as such, we can eventually put one near the bar at roundstart like it was on Aurora (the station).

The old piano and violin implementations, along with their sounds, are removed from the game as requested by Matt.

### Asset Licenses
The following assets that have not been created by myself are included in this PR:

All the assets that were added are, to my knowledge, unmodified copy from Baystation12 which is released under the GNU Affero license as described in the Baystation 12 repository.

### CODE REVIEW

**Comment by FluffyGhoster** [comment]
Cargo items when merged/testmerged to add (I hope I did everything correctly):

`name;supplier;description;categories;price;items;item_mul;access;container_type;groupable;order_by;error_message
polyguitar;nt;A polyguitar, better than a plain guitar.;"[""supply"",""hospitality""]";200;"{""PolyGuitar Instrument"":{""path"":""/obj/item/device/synthesized_instrument/guitar/multi"",""vars"":[]}}";1;0;box;1;\N;\N
violin;nt;An old violin.;"[""supply"",""hospitality""]";250;"{""Violin Instrument"":{""path"":""/obj/item/device/synthesized_instrument/violin"",""vars"":[]}}";1;0;box;1;\N;\N
trumpet;nt;An old trumpet.;"[""supply"",""hospitality""]";300;"{""Trumpet Instrument"":{""path"":""/obj/item/device/synthesized_instrument/trumpet"",""vars"":[]}}";1;0;box;1;\N;\N
piano;nt;A piano;"[""hospitality""]";1200;"{""Piano Instrument"":{""path"":""/obj/structure/synthesized_instrument/synthesizer/piano"",""vars"":[]}}";1;0;crate;0;\N;\N
sound synthesizer;nt;A sound synthesizer;"[""hospitality""]";1900;"{""Sound Synthesizer Instrument"":{""path"":""/obj/structure/synthesized_instrument/synthesizer"",""vars"":[]}}";1;0;crate;0;\N;\N`

**Comment by FluffyGhoster** [comment]
!review

File: code/_helpers/maths.dm [hunk 1]
```diff
@@ -28,6 +28,9 @@
 /proc/Floor(x)
 	return round(x)
 
+/// Value or the next integer in a positive direction: Ceil(-1.5) = -1 , Ceil(1.5) = 2
+#define Ceil(value) ( -round(-(value)) )
+
```
Comment (NonQueueingMatt) [inline code/_helpers/maths.dm:33]: this is equivalent to Ceiling()

File: code/modules/synthesized_instruments/real_instruments/Piano/piano.dm [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/obj/structure/synthesized_instrument/synthesizer/piano
+	name = "space piano"
+	desc = "A surprisingly high-tech piano with a digital display for modifying sound output."
+	icon_state = "piano"
+	path = /datum/instrument/piano
+
+// /obj/random_multi/single_item/piano
+// 	name = "piano spawnpoint"
+// 	id = "spacepiano"
+// 	item_path = /obj/structure/synthesized_instrument/synthesizer/piano
```
Comment (NonQueueingMatt) [inline code/modules/synthesized_instruments/real_instruments/Piano/piano.dm:10]: commented code

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

File: code/__defines/_macros.dm [hunk 3]
```diff
@@ -20,6 +20,7 @@
 #define SPAN_SUBTLE(X) ("<span class='subtle'>" + X + "</span>")
 #define SPAN_SOGHUN(X) ("<span class='soghun'>" + X + "</span>")
 #define SPAN_VOTE(X) ("<span class='vote'>" + X + "</span>")
+#define SPAN_CLASS(class, X) "<span class='[class]'>[X]</span>"
```
Comment (NonQueueingMatt) [inline code/__defines/_macros.dm:23]: this is a duplicate of span(class, text) above

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Review by Alberyk — APPROVED** [review]

**Pull request merged**